test/util/IdTableHelpers.cpp

```diff
@@ -60,28 +67,27 @@ void sortIdTableByJoinColumnInPlace(IdTableAndJoinColumn& table) {
 // ____________________________________________________________________________
 IdTable generateIdTable(
     const size_t numberRows, const size_t numberColumns,
-    const std::function<IdTable::row_type()>& rowGenerator) {
+    const std::function<std::vector<ValueId>()>& rowGenerator) {
   AD_CONTRACT_CHECK(numberRows > 0 && numberColumns > 0);
 
   // Creating the table and setting it to the wanted size.
   IdTable table{numberColumns, ad_utility::testing::makeAllocator()};
   table.resize(numberRows);
 
   // Fill the table.
-  std::ranges::generate(table, [&numberColumns, &rowGenerator]() {
-    // Make sure, that the generated row has the right size, before passing it
-    // on.
-    IdTable::row_type row = rowGenerator();
-
-    if (row.numColumns() != numberColumns) {
-      throw std::runtime_error(absl::StrCat(
-          "The row generator generated a IdTable::row_type of size ",
-          row.numColumns(), ", which is not the wanted size ", numberColumns,
-          "."));
-    }
-
-    return row;
-  });
+  for (auto restrictedRow = table.begin(); restrictedRow != table.end();
+       restrictedRow++) {
+    // Make sure, that the generated row has the right
+    // size, before using it.
+    std::vector<ValueId> generatedRow = rowGenerator();
+    AD_CONTRACT_CHECK(generatedRow.size() == numberColumns);
+
+    // The iterators of `IdTable` never directly give write access. Instead you
+    // have to cast them, before writting to them.
+    std::ranges::copy(
+        std::move(generatedRow),
+        static_cast<IdTable::row_reference>(*restrictedRow).begin());
+  }
```

Comment:
I now see your struggles with the missing interfaces.
I think `std::generate` with the `row_type` would have worked.
But I think you can just `reserve` instead of `resize` the table, 
And then simply call `push_back()` for the `IddTable`, which should accept a `vector<ValueId>`. That way you don't need to deal with these strange iterator types as much. I will nevertheless look into them, because I really want this safer behavior to be there.
However also look into the following:
temporaries of the row references should always work, so something like
(*restrictedRow).begin() should work, but it probably doesn't because an overload of begin is missing etc.
but again, simply use `reserve` and `push_back` and life is fine.

Reply:
> I think `std::generate` with the `row_type` would have worked. 

Oh, it does. I simply didn't want to create a temporary `row_type`, if I don't have to. More performant and all that.

> And then simply call `push_back()` for the `IddTable`, which should accept a `vector<ValueId>`. That way you don't need to deal with these strange iterator types as much.

Unfortunately, you can't implicitly convert a `std::vector` to a initializer list. So that doesn't work.

> However also look into the following: temporaries of the row references should always work, so something like (*restrictedRow).begin() should work, but it probably doesn't because an overload of begin is missing etc. but again, simply use `reserve` and `push_back` and life is fine.

I'm not really sure, what you are talking about here. As far as I know, `(*restrictedRow).begin() ` works perfectly fine. I don't use it, because read only access is useless to me in this situation.

Reply:
Okay,
This is really insightful. 
Okay:
I imagined there was a `push_back` that takes a vector at least for the dynamic `IdTable`, but there is not. 
If you want to, you can write it (here or in a separate PR). Best you accept a `std::span<const T>`, that is the most generic interface that a vector etc. can use.

I meant that `(*restrictedRow).begin()` should give you a non-const iterator (temporaries like `*iterator` should have write access as long as you don't bind them to an `auto` variable. But maybe that's not the case, I'll look into it.

Reply:
Should be better now: [0b402ae](https://github.com/ad-freiburg/qlever/pull/1016/commits/0b402aeb603f6e22aeecee9f3bc881b6f94d5df7)